lib/extensions/user_notifications_extension.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module DiscourseChat::UserNotificationsExtension
+  def chat_summary(user, opts)
+    @email_statuses = ChatMessageEmailStatus
+      .includes(chat_message: [:chat_channel, :user])
+      .where(user: user, status: ChatMessageEmailStatus::STATUSES[:unprocessed])
+      .to_a
+      .filter { |status| status.chat_message.present? }
+    return unless @email_statuses.any?
+
+    build_summary_for(user)
+    opts = {
+      from_alias: I18n.t('user_notifications.chat_summary.from', site_name: Email.site_title),
+      subject: I18n.t('user_notifications.chat_summary.subject', count: @email_statuses.count, email_prefix: @email_prefix, date: short_date(Time.now)),
+      add_unsubscribe_link: true,
+      unsubscribe_url: "#{Discourse.base_url}/email/unsubscribe/#{@unsubscribe_key}",
+    }
+
+    ChatMessageEmailStatus.where(id: @email_statuses.map(&:id)).update_all(status: ChatMessageEmailStatus::STATUSES[:processed])
```

Comment:
This feels like the wrong place to update all email status summaries. Could we do this in the job but _after_ having sent out the email? And maybe it would suffice to query by `user_id`, surely we want to mark as processed all existing email message statuses for said user?